components/ILIAS/Authentication/classes/Password/class.ilLocalPasswordSettingsGUI.php

```diff
@@ -0,0 +1,249 @@
+    private UIFactory $ui_factory;
+    private UIRenderer $ui_renderer;
+    private ilGlobalTemplateInterface $tpl;
+    private ilLanguage $lng;
+    private ilObjUser $user;
+    private ilCtrlInterface $ctrl;
+    private ilLocalPasswordManager $password_manager;
+
+    public function __construct()
+    {
+        global $DIC;
+        $this->user = $DIC->user();
+        $this->ctrl = $DIC->ctrl();
+        $this->lng = $DIC->language();
+        $this->tpl = $DIC->ui()->mainTemplate();
+        $this->request = $DIC->http()->request();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->ui_renderer = $DIC->ui()->renderer();
+        $this->password_manager = ilLocalPasswordManager::getInstance();
+        $this->lng->loadLanguageModule('user');
+    }
+
+    public function executeCommand(): void
+    {
+        $cmd = $this->ctrl->getCmd();
+        switch ($cmd) {
+            default:
+                $this->$cmd();
+
+                break;
+        }
+    }
+
+    public function showPassword(
+        Form $form = null,
+        bool $hide_form = false
+    ): void {
+        // check whether password of user have to be changed
+        // due to first login or password of user is expired
+        if ($this->user->isPasswordChangeDemanded()) {
+            $this->tpl->setOnScreenMessage('info', $this->lng->txt('password_change_on_first_login_demand'));
+        } elseif ($this->user->isPasswordExpired()) {
+            $msg = $this->lng->txt('password_expired');
+            $password_age = $this->user->getPasswordAge();
+            $this->tpl->setOnScreenMessage('info', sprintf($msg, $password_age));
+        }
+
+        if (!$form && !$hide_form) {
+            $form = $this->getPasswordForm();
+        }
+        $this->tpl->setContent(!$hide_form ? $this->ui_renderer->render($form) : '');
+        $this->tpl->printToStdout();
+    }
+
+    public function getPasswordForm(
+        ServerRequestInterface $request = null,
+        array $errors = []
+    ): Form {
+        $items = [];
+        if ($this->password_manager->allowPasswordChange($this->user)) {
+            $pw_info_set = false;
+            if ((int) $this->user->getAuthMode(true) === ilAuthUtils::AUTH_LOCAL) {
+                $cpass = $this->ui_factory->input()->field()->password(
+                    $this->lng->txt('current_password'),
+                    ilSecuritySettingsChecker::getPasswordRequirementsInfo()
+                );
+
+                $pw_info_set = true;
+                if ($this->user->getPasswd()) {
+                    $cpass = $cpass->withRequired(true);
+                }
+                $cpass = $cpass->withRevelation(true);
+                $cpass_error = $errors['current_password'] ?? [];
+                if ($cpass_error !== []) {
+                    $cpass = $cpass->withError(implode('<br>', $cpass_error));
+                }
+
+                $items['current_password'] = $cpass;
```

Comment:
Could we add `withAdditionalTransformation` for the validations you currently do in `savePassword`?